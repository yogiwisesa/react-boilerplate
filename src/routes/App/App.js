import React, { Component } from 'react';
import { connect } from 'react-redux';
import { SetSampleAction, FetchAPISample } from '../../redux/actions';
import { Alert } from "react-bootstrap";
import logo from './logo.svg';

import './App.scss';
export class App extends Component {
  
  componentDidMount() {
    this.props.SetSampleAction({'Lorem': 'lorem'})
    this.props.FetchAPISample()
  }

  render() {
    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <p>
            Edit <code>src/App.js</code> and save to reload.
          </p>
          <Alert variant="primary">Hello, here is a bootstrap alert!</Alert>
          <a
            className="App-link"
            href="https://reactjs.org"
            target="_blank"
            rel="noopener noreferrer"
          >
            Learn React
          </a>
        </header>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  SampleReducer: state.SampleReducer
});

const mapDispatchToProps = {
  SetSampleAction,
  FetchAPISample
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(App);