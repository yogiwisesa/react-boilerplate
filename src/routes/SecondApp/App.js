import React, { Component } from 'react';
import { connect } from 'react-redux';
import { SetSampleAction, FetchAPISample } from '../../redux/actions';
import { Alert } from "react-bootstrap";
import logo from './logo.svg';

import './App.scss';
export class SecondApp extends Component {
  
  componentDidMount() {
    this.props.SetSampleAction({'Lorem': 'lorem'})
    this.props.FetchAPISample()
  }

  render() {
    return (
      <div className="App">
        <header className="App-header">
          <h1>This is the second page</h1>
          <img src={logo} className="App-logo" alt="logo" />
          <p>
            Edit <code>src/App.js</code> and save to reload.
          </p>
          <Alert variant="primary">Hello, here is a bootstrap alert!</Alert>
          <a
            className="App-link"
            href="https://reactjs.org"
            target="_blank"
            rel="noopener noreferrer"
          >
            Learn React
          </a>
        </header>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  SampleReducer: state.SampleReducer
});

const mapDispatchToProps = {
  SetSampleAction,
  FetchAPISample
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(SecondApp);