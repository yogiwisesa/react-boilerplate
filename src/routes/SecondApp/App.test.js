import React from 'react';
import { shallow } from 'enzyme';
import { App } from './App';

let wrapper;
beforeEach(() => {
  wrapper = shallow(<App/>);
});

it('Should render correctly', () => {
  expect(wrapper).toMatchSnapshot();
});

it('Should App-header class', () => {
  expect(wrapper.find('.App-header').length).toBeGreaterThan(0);
});
