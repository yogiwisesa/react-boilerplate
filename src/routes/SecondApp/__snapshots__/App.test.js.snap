// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render correctly 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <p>
      Edit 
      <code>
        src/App.js
      </code>
       and save to reload.
    </p>
    <PickerWrapper
      locale={Object {}}
      onChange={[Function]}
      onOk={[Function]}
      onOpenChange={[Function]}
      popupStyle={Object {}}
      transitionName="slide-up"
    />
    <Button
      block={false}
      ghost={false}
      loading={false}
    >
      Hello World
    </Button>
    <a
      className="App-link"
      href="https://reactjs.org"
      rel="noopener noreferrer"
      target="_blank"
    >
      Learn React
    </a>
  </header>
</div>
`;
